samtranslator/translator/arn_generator.py

```diff
@@ -3,12 +3,18 @@
 class ArnGenerator(object):
 
     @classmethod
-    def generate_arn(cls, partition, service, resource):
+    def generate_arn(cls, partition, service, resource, include_account_id=True):
         if not service or not resource:
             raise RuntimeError("Could not construct ARN for resource.")
 
-        return 'arn:{0}:{1}:${{AWS::Region}}:${{AWS::AccountId}}:{2}'.format(partition, service, resource)
+        arn = 'arn:{0}:{1}:${{AWS::Region}}:'
 
+        if include_account_id:
```

Comment:
Which resources don't require an account ID? I noticed that this is only used for `authorizer_uri`, do you have a link for that documentation handy? If not I'll look for it and read up on it.

Reply:
Note sure. This was added for reusability.